File: client/src/app/interfaces/question.ts
import { Choice } from '@app/interfaces/choice';
import { EstimatedParameters } from '@app/interfaces/estimated-parameters';

export interface Question {
    id: string;
    type: string;
    text: string;
    points: number;
    choices?: Choice[];
    answer?: string;
    estimatedParameters?: EstimatedParameters;
    lastModification: string;
    pictureUrl: string;
    pictureFile?: File | null; // To keep the file for uploading question picture AFTER the game / bankQuestion has been validated.
}

File: client-mobile/app/src/main/java/com/example/polyquiz/Question.kt
package com.example.polyquiz

data class Question(
    val id: String,
    val type: String,
    val text: String,
    val points: Double,
    val choices: List<Choice>? = null,
    val answer: String? = null,
    val lastModification: String
)

data class Choice(
    val text: String,
    val isCorrect: Boolean? = null,
)
